dipy/tracking/streamline.py

```diff
@@ -312,21 +353,23 @@ def orient_by_rois(streamlines, roi1, roi2, affine=None, copy=True):
         roi1 = apply_affine(affine, roi1)
         roi2 = apply_affine(affine, roi2)
 
-    # Make a copy, so you don't change the output in place:
-    if copy:
-        new_sl = deepcopy(streamlines)
-    else:
-        new_sl = streamlines
+    if as_generator:
+        if out is not None:
+            w_s = "Cannot set output, when input is a generator"
+            raise ValueError(w_s)
+        return _orient_generator(streamlines, roi1, roi2)
 
-    for idx, sl in enumerate(streamlines):
-        dist1 = cdist(sl, roi1, 'euclidean')
-        dist2 = cdist(sl, roi2, 'euclidean')
-        min1 = np.argmin(dist1, 0)
-        min2 = np.argmin(dist2, 0)
-        if min1[0] > min2[0]:
-            new_sl[idx] = sl[::-1]
-
-    return new_sl
+    else:
+        # If it's a generator on input, we may as well generate it
+        # here and now:
+        if isinstance(streamlines, types.GeneratorType):
+            out = list(streamlines)
+        elif out is None:
+            # Make a copy, so you don't change the output in place:
+            out = deepcopy(streamlines)
+        else:
+            out = streamlines
```

Comment:
On further thought, I think that an `out` kwarg doesn't make sense. It would have to be a list of exactly the same length as the original streamlines. May as well have an `inplace` kwarg that would just do that.